src/ray/raylet/node_manager.cc

```diff
@@ -300,26 +301,30 @@ void NodeManager::ProcessNewClient(LocalClientConnection &client) {
 
 void NodeManager::DispatchTasks() {
   // Work with a copy of scheduled tasks.
-  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  const auto &scheduled_tasks = local_queues_.GetScheduledTasks();
   // Return if there are no tasks to schedule.
   if (scheduled_tasks.empty()) {
     return;
   }
   const ClientID &my_client_id = gcs_client_->client_table().GetLocalClientId();
+  const auto &local_resources =
```

Comment:
@atumanov @eric-jj 

There may be a bug here (or at least a change in behavior). Before, we called `AssignTask` inside the `for` loop, which presumably modified `local_resources`. However, now we call `AssignTask` at the very end, so the check `if (!task_resources.IsSubset(local_resources)) {` may give different results now.

I will try reverting some of this and see if that makes the problem I'm seeing go away.

Reply:
I will investigate it.

Reply:
@robertnishihara good catch. That's the reason why I didn't batch `AssignTask` calls before. 
`this->cluster_resource_map_[my_client_id].Acquire(spec.GetRequiredResources()));` <-- this line acquires resources when the task is assigned. This makes sure that the locally available resources are updated at each for loop iteration.

Reply:
yes. you are right, I have reverted it back.